include/gatekeeper_ggu.h

```diff
@@ -47,35 +47,77 @@ struct ggu_config {
 	struct mailbox    mailbox;
 };
 
+/* Enumeration of policy decisions the GT-GK block can process. */
+enum {
+	/* Decline an IPv4 flow. */
+	GGU_DEC_IPV4_DECLINED = 0,
+	/* Decline an IPv6 flow. */
+	GGU_DEC_IPV6_DECLINED,
+	/* Grant an IPv4 flow. */
+	GGU_DEC_IPV4_GRANTED,
+	/* Grant an IPv6 flow. */
+	GGU_DEC_IPV6_GRANTED,
+	__MAX_GGU_DEC
+};
+
+/* On the wire policy decision to be processed by the GT-GK Unit. */
+struct ggu_decision {
+	/* The policy decision type. */
+	uint16_t type;
+	/* The length of the policy decision, including any parameters. */
+	uint16_t len;
+
+	/* The IP flow relevant to this policy decision. */
+	uint8_t  ip_flow[0];
+
+	/* Parameters for this policy decision would follow the IP flow. */
+}__attribute__((packed));
+
 /*
- * Since the length of IPv6 policy is much larger than IPv4 policy,
- * to save network bandwidth, we choose to use different data structures
- * to store the in-packet policies for IPv4 and IPv6, respectively.
+ * Packets that flow between the GT block and the GT-GK Unit.
+ * 
+ * Packet format: Ethernet header(s) + IP header + UDP header + Data.
+ * The UDP payload data is of the following format:
  *
- * Packet format: Ethernet headers + IP header + UDP header + Data.
- * In the UDP payload, the following format would save a lot of bytes
- * when there are decline decisions:
- *  v1, n1, n2, n3, n4: Each of these fields are 1-byte long.
- *  v1 is a constant indicating the version of the format, in this case 1.
- *  n1 is the number of IPv4 decline decisions.
- *  n2 is the number of IPv6 decline decisions.
- *  n3 is the number of IPv4 granted decisions.
- *  n4 is the number of IPv6 granted decisions.
+ *  v1: a constant indicating the version of the format, in this case 1
+ *  reserved: reserved space to keep the fields 32-bit aligned
+ *  [policy decisions]: a list of one or more policy decisions
```

Comment:
Maybe ``[policy decision(s)]``?

Style: add a period at the end of the sentence.